.github/workflows/nightly-emscripten.yml

```diff
@@ -0,0 +1,96 @@
+name: Emscripten nightly build + push to solc-bin
+
+on:
+  schedule:
+    # Run once a day, at midnight
+    - cron: '0 0 * * *'
+
+env:
+  TARGET_BRANCH: gh-pages
+  COMMITTER_NAME: emscripten nightly action
+  COMMITTER_EMAIL: builds@ethereum.org
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  build-emscripten-nightly:
+    runs-on: ubuntu-latest
+    outputs:
+      nightly-version: ${{ env.NIGHTLY_VERSION }}
+      matching-nightlies-in-the-repo: ${{ env.MATCHING_NIGHTLIES_IN_THE_REPO }}
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          repository: 'ethereum/solidity'
+          ref: 'develop'
+          path: 'solidity/'
+
+      - name: Clone solc-bin repository without checking out a working copy
+        run: |
+          git clone --no-checkout --branch "$TARGET_BRANCH" "https://github.com/${GITHUB_REPOSITORY}.git" solc-bin/
+
+          # For some reason git stages all files for deletion when you use --no-checkout
+          cd solc-bin/
+          git reset HEAD --quiet
+
+      - name: Check if there's already a nightly with the same date or commit ID
+        run: |
+          cd solidity/
+          solidity_version=$("scripts/get_version.sh")
+          last_commit_timestamp=$(git log -1 --date=iso --format=%ad HEAD)
+          last_commit_date=$(date --date="$last_commit_timestamp" --utc +%Y.%-m.%-d)
+          last_commit_hash=$(git rev-parse --short=8 HEAD)
+
+          cd ../solc-bin/
+          matching_nightlies_in_the_repo="$(
+            git ls-files "bin/soljson-v${solidity_version}-nightly.${last_commit_date}+commit.*.js";
+            git ls-files "bin/soljson-v${solidity_version}-nightly.*+commit.${last_commit_hash}.js"
+          )"
+          nightly_version="v${solidity_version}-nightly.${last_commit_date}+commit.${last_commit_hash}"
+
+          echo "::set-env name=NIGHTLY_VERSION::${nightly_version}"
+          echo "::set-env name=MATCHING_NIGHTLIES_IN_THE_REPO::${matching_nightlies_in_the_repo}"
+
+      - name: Build soljson.js
+        if: "!env.MATCHING_NIGHTLIES_IN_THE_REPO"
+        run: |
+          cd solidity/
+          # Note that this script will spawn and build inside a docker image (which works just fine in github actions).
+          scripts/build_emscripten.sh
+
```

Comment:
Should we maybe run at least a simple test?

Reply:
Done. I added third job that runs `solc-js` tests on the nightly according to @ekpyron's suggestion.